File: api/webhook.js
const express = require("express");
const serverless = require("serverless-http"); // permite rodar no Vercel
const app = express();

app.use(express.json());

const verifyToken = process.env.VERIFY_TOKEN ;

// GET para verificação do webhook
app.get("/", (req, res) => {
  const {
    "hub.mode": mode,
    "hub.challenge": challenge,
    "hub.verify_token": token,
  } = req.query;

  if (mode === "subscribe" && token === verifyToken) {
    console.log("WEBHOOK VERIFIED");
    return res.status(200).send(challenge);
  }
  return res.status(403).end();
});

// POST para receber eventos
app.post("/", (req, res) => {
  const timestamp = new Date().toISOString().replace("T", " ").slice(0, 19);
  console.log(`Webhook received ${timestamp}`);
  console.log(JSON.stringify(req.body, null, 2));
  res.status(200).end();
});

// **Não usar app.listen() no Vercel**
module.exports = app.listen(3000, () => {
  console.log("Server is running on port 3000");
});

// Exporta o app para o Vercel
module.exports.handler = serverless(app);

File: api/webhook.py
import json
import os

VERIFY_TOKEN = os.getenv("VERIFY_TOKEN", "teste123")


def handler(request):
    # GET → verificação
    if request.method == "GET":
        mode = request.args.get("hub.mode")
        challenge = request.args.get("hub.challenge")
        token = request.args.get("hub.verify_token")

        if mode == "subscribe" and token == VERIFY_TOKEN:
            return challenge, 200

        return "Forbidden", 403

    # POST → eventos
    if request.method == "POST":
        try:
            payload = request.get_json()
            print("Webhook recebido:")
            print(json.dumps(payload, indent=2))
        except Exception as e:
            print("Erro ao ler payload:", e)

        return "OK", 200

    return "Method Not Allowed", 405
